doc/nrf/gs_modifying.rst

```diff
@@ -84,55 +93,125 @@ The following CMake commands can be managed by SES, if they target the ``app`` l
 
 The :file:`CMakeLists.txt` files for the sample applications in the :file:`sdk-nrf` repository already have the required tags.
 Therefore, if you always use SES to maintain them, you do not need to worry about tagging.
-Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c` file as source::
+Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c` file as source:
+
+.. code-block:: c
 
    # NORDIC SDK APP START
    target_sources(app PRIVATE src/main.c)
    # NORDIC SDK APP END
 
+Advanced compiler settings
+==========================
+
+The application has full control over the build process.
+
+Using Zephyr's configuration options is the standard way of controlling how the system is built.
+These options can be found under Zephyr's menuconfig :guilabel:`Build and Link Features` > :guilabel:`Compiler Options`.
+For example, to turn off optimizations, select :kconfig:`CONFIG_NO_OPTIMIZATIONS`.
+
+Compiler options not controlled by the Zephyr build system can be controlled through the :kconfig:`CONFIG_COMPILER_OPT` Kconfig option.
+
 .. _configure_application:
 
 Configuring your application
 ****************************
 
-If your application uses a provided library or targets a specific board, you might want to change the default configuration of the library or board.
+You might want to change the default options of the application.
 There are different ways of doing this, but not all will store your configuration permanently.
 
-The default configuration for a library is specified in its :file:`Kconfig` file.
-Similarly, the default configuration for a board is specified in its :file:`*_defconfig` file (and its :file:`Kconfig.defconfig` file, see :ref:`zephyr:default_board_configuration` in the Zephyr documentation for more information).
-The configuration for your application, which might override some default options of the libraries or the board, is specified in a :file:`prj.conf` file in the application directory.
+.. _configuration_system_overview:
 
-For detailed information about configuration options, see :ref:`zephyr:application-kconfig` in the Zephyr documentation.
+Configuration system overview
+=============================
```

Comment:
The two sentences in "Configuring your application" are not saying anything useful. And it is a bit ominous to mention a danger without an immediate follow up about how to mitigate the danger.

Can we promote "Configuration system overview" up one title level and delete "Configuring your application" ?